### DISCUSSION TITLE
Must specify either pores or throats

### DISCUSSION THREAD

**Discussion by NBL2** [PMEAL/OpenPNM#2742] [Q&A]
Dear OpenPNM Community,

I hope this message finds you well. I am currently working on a project that involves using the OpenPNM package to model transport in porous media. However, I've run into an issue with the `rate()` method in the `algorithms_transport.py` module, and I was wondering if anyone in the community might be able to help me troubleshoot this issue.
![image](https://user-images.githubusercontent.com/131282142/233132152-6d889b08-a45c-4d15-b0f4-cea7d3ab6578.png)
Specifically, when I try to run my code, I receive the following error message: "Must specify either pores or throats." I've checked my input arguments and consulted the documentation, but I'm still having trouble figuring out what's going wrong. I would be incredibly grateful if anyone in the community could offer guidance or suggestions on resolving this error.
![image](https://user-images.githubusercontent.com/131282142/233133281-15724620-d07b-4f45-b3e4-6439828cd31b.png)

albeit I have specified pores but `Q = sf.rate(pores=pn.pores('xmin'),mode ='group')`  it keeps asking me for the specification

Thank you for taking the time to read my message, and I appreciate any help or advice that anyone might be able to offer.

Best regards,

**Comment by jgostick** [comment]
Your code looks fine, so not sure what's the issue.  Can you do ``print(pn.pores('xmin'))`` and see what you get?  The message is indicating that this returns `[]` (i.e. an empty list).  I see that 191 pores are supposed to be labeled as `'xmin'`, but maybe something is amiss.

**Reply by NBL2** [reply]
Did as you said and got this result...
![image](https://user-images.githubusercontent.com/131282142/233183640-4a30faf3-89b3-4e7d-a98a-c2cead9532d6.png)

As expected when I ran the line of code below
![image](https://user-images.githubusercontent.com/131282142/233184330-4d1f879b-5e0f-45a3-b9f6-c48841f15423.png)
it trimmed of the pores in all the directions, 
![image](https://user-images.githubusercontent.com/131282142/233184230-2d394f34-bdc3-4175-93de-8b179996d983.png)

but if I don't do this step then another error prevails
![image](https://user-images.githubusercontent.com/131282142/233184836-26b09434-1021-4746-8882-87c5f04d80cf.png)
could you kindly tell me what could be the possible way forward?

thanks for your reply

**Reply by jgostick** [reply]
the health function returns a list of 'suggested' pore to disconnect. This suggestion is not always correct.  You're going to have to dig a little deeper about what is going on in your case. [This example](https://openpnm.org/examples/reference/networks/managing_clustered_networks.html) might help get you started.

**Reply by NBL2** [reply]
thank you for looking into my case. I'll have a look at it and try to understand the problem thoroughly. thank you again for your reply

**Reply by NBL2** [reply]
Pardon me sir @jgostick , would you please tell me if the snow algorithm is only applicable on equal dimensions, because i tried it on a 950x950x550 and it took an hour but didnot stop computing so I had to force stop it but when I did it on 550x550x550, it gave me results in a few minutes.

**Reply by jgostick** [reply]
There is no limitation on the shape of the image, but your numbers are concerning.  Can you try something like 950x500x550?  My first hunch is that your PC is running out of ram on the larger image and is needed to read/write from the disk, which is very slow.

**Reply by NBL2** [reply]
I humbly thank you for your reply.. The issue was I was doing 950x950x550 whereas it should be 550x950x950 while using reshape function. Thank you sir @jgostick

**Reply by jgostick** [reply]
ahh, so when reshaping with was turning the image into a jumbled mess and therefore taking ages.  Glad you got it working.
